gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -99,6 +100,11 @@
     this.locationLockState =
         LiveDataReactiveStreams.fromPublisher(
             locationLockStateFlowable.startWith(BooleanOrError.falseValue()));
+    this.iconTint =
+        LiveDataReactiveStreams.fromPublisher(
+            locationLockStateFlowable
+                .map(locked -> locked.isTrue() ? R.color.colorMapBlue : R.color.colorGrey800)
+                .startWith(R.color.colorGrey800));
```

Comment:
Can we move this to the `MapContainerFragment`? Only the true/false should be exposed and then the color can be set using data binding. We can also get rid of the BindingAdapter then. Wdyt?

Reply:
How would we use data binding without the BindingAdapter? What would that look like?

Also note that we aim to have as little logic as possible in the View components (MapContainerFragment) to improve testability, so moving this in this into the Fragment might be consider a move in the opposite direction. Open to ideas though!

Reply:
On second thought, let's stick with the current approach. Sorry for the confusion!